tests/integration/controller/mandatorymodule/deletion/mandatory_modules_deletion_controller_test.go

```diff
@@ -0,0 +1,159 @@
+package mandatory_test
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	compdescv2 "github.com/open-component-model/ocm/pkg/contexts/ocm/compdesc/versions/v2"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/kyma-project/lifecycle-manager/pkg/testutils/builder"
+
+	. "github.com/kyma-project/lifecycle-manager/pkg/testutils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var ErrNoMandatoryManifest = errors.New("manifest for mandatory Module not found")
+
+const (
+	mandatoryChannel = "dummychannel"
+)
+
+var _ = Describe("Mandatory Module Deletion", Ordered, func() {
+	Context("Given Kyma with one mandatory Module Manifest CR on Control-Plane", func() {
+		kyma := NewTestKyma("no-module-kyma")
+		registerControlPlaneLifecycleForKyma(kyma)
+
+		It("Then Kyma CR should result in a ready state and mandatory manifest is created", func() {
+			Eventually(KymaIsInState).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, shared.StateReady).
+				Should(Succeed())
+			Eventually(ManifestExistsWithAnnotation).
+				WithContext(ctx).
+				WithArguments(controlPlaneClient, shared.FQDN, "kyma-project.io/template-operator").
+				Should(Succeed())
+		})
+
+		It("And mandatory finalizer is added to the mandatory ModuleTemplate", func() {
+			Eventually(mandatoryModuleTemplateFinalizerExists).
+				WithContext(ctx).
+				WithArguments(controlPlaneClient, client.ObjectKey{
+					Namespace: apimetav1.NamespaceDefault,
+					Name:      "mandatory-module",
+				}).
+				Should(Succeed())
+		})
+
+		It("When mandatory ModuleTemplate marked for deletion", func() {
+			Eventually(deleteMandatoryModuleTemplates).
+				WithContext(ctx).
+				WithArguments(controlPlaneClient).
+				Should(Succeed())
+		})
+
+		It("Then mandatory Manifest is deleted", func() {
+			Eventually(ManifestExistsWithAnnotation).
+				WithContext(ctx).
+				WithArguments(controlPlaneClient, shared.FQDN, "kyma-project.io/template-operator").
+				Should(Not(Succeed()))
+		})
+
+		It("And finalizer is removed from mandatory ModuleTemplate", func() {
```

Comment:
I think we should conform to a common style in writing these tests.
Take a look at `module_deletion_test.go` - I consider this to be the "canonical" one with respect to the test structure.
The "And" blocks there are realized by a nested `By("...")` calls, not as `It()` containers.